### DISCUSSION TITLE
TerriaJS Localization ?

### DISCUSSION THREAD

**Discussion by arky** [TerriaJS/terriajs#6213] [Ideas]
This is general discussion on future localization support in TerriaJS.  We have a number of complete and partial localization efforts on (Weblate platform)[https://hosted.weblate.org/projects/terriajs/terriajsnext/#translations] however the strings have not yet landed in TerriaJS repo. 

Curious to discuss and learn more about TerriaJS Localization roadmap @terriaJS team's @kring @steve9164 @meh9 @nf-s  @KeyboardSounds @AnaBelgun 

You can read the discussion on Localization Web platform here:
https://github.com/TerriaJS/terriajs/discussions/5072

**Comment by tephenavies** [comment]
Hi @arky. Thanks for reaching out. The translation files are present in terriajs releases, however it's currently hard to use them (I think it requires forking terriajs). I've proposed this PR https://github.com/TerriaJS/terriajs/pull/6219 to improve translation file fetching and i18n configuration. Let me know what you think.

**Reply by arky** [reply]
Thanks @steve9164 for the response.  

Looking at how other projects pull and manage their localization. The most common workflow is adding run script / command in package.json to pull/update localization files and add them to TerriaJS. 

A good start is to identify languages that  are 1) fully localized  and 2) have core contributors keeping the strings update between releases and pull them regularly into main repo using weblate integration . This automation has added advantage of motivating localization contributors see their work shipped with the project.

**Reply by tephenavies** [reply]
https://github.com/TerriaJS/terriajs/pull/6219 has been merged and it's now much easier to add any language with strings from weblate! I'm trialling a few languages in http://ci.terria.io/main/ (currently it's using languages that were above 40% translated on 2022/06/01). In future we may include the following languages in TerriaMap by default:
* English
* Japanese
* Italian
* French

and maybe:
* Catalan
* Czech
* Portuguese (Brazil)

**Reply by tephenavies** [reply]
BTW configuration for languages is like below (except put this as JSON in "parameters" in your config.json file)

https://github.com/TerriaJS/terriajs/blob/bd6198e66b9cf4314a95464acba8ab5d40516ef8/buildprocess/ci-values.yml#L64-L77

**Comment by arky** [comment]
Thank you @steve9164 for all the work!
